.github/workflows/ci-app.yml

```diff
@@ -1,28 +1,79 @@
-name: NextJS CI
+name: React CI
 
 on:
   push:
     branches:
       - main
   pull_request:
+    paths:
+      - app/**
+      - .github/workflows/ci-app.yml
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    working-directory: ./app
 
 jobs:
   test:
-    name: pass/fail checks
+    name: Test, Lint, Type check
```

Comment:
thoughts on making these separate jobs? i personally find it useful to know what failed, was it the tests or the linter? it doesn't seem like one should depend on the other succeeding

Reply:
<img width="938" alt="CleanShot 2023-03-03 at 19 51 01@2x" src="https://user-images.githubusercontent.com/371943/222874456-d738cb8c-9147-44f1-850e-d76430350fdb.png">